### DISCUSSION TITLE
What is the maximum response body size that FastAPI should be able to gracefully handle?

### DISCUSSION THREAD

**Discussion by jasonyang101** [fastapi/fastapi#11246] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
N/A. Will add later on if deemed necessary.
```

### Description

We have a request that queries a datastore and then returns the values to the client. We've seen a ton of latency issues on this API, and so as part of the debugging, I added a log statement (Statement A) as the final line before returning the response in the path function of the defined API. 

I then have another log statement (Statement B) in some middleware that logs the total request time for this route. My problem is that often times, the amount of time in between these statements A and B can be quite long (from 7 seconds to 30 seconds) To be clear, this difference is between the final line of code of the API, and the first line of the middleware, which logs the route and the time it took to process the request. The middleware which logs statement B is a descendant of BaseHTTPMiddleware. Notedly, our path operation is defined via synchronously, via `def`. 

We do not have any blocking operations in our middleware as of now, and so I'm unclear as to where this delay may be coming from. One thing to note is that admittedly, the response is sizable, but nothing insane - 35MB total (although, let me know if you think it _is_ insane).

The main question: At what point should I think that this size is just too large and I need to implement pagination and a max size on my API? 

Follow up question: We previously incorrectly had APIs defined as async (`async def`) and were using Uvicorn workers, but it was incorrect because our DB library was the blocking version of sqlalchemy. We've switched these to synchronous to match our db query in the short term, but now I'm concerned about our worker class. What happens when we have `Uvicorn` workers, which seem to be optimal for ASGI, but synchronously defined endpoints? I see very little information about this in the docs.

### Operating System

Linux

### FastAPI Version

0.99.1

### Pydantic Version

1.10.12

### Python Version

3.11.5

**Comment by jgould22** [comment, marked as answer]
35mb is a lot of JSON to parse and validate.

If you are defining a response object I suggest instead you avoid the response validation and return a response directly.

https://fastapi.tiangolo.com/tutorial/response-model/#return-a-response-directly

https://fastapi.tiangolo.com/advanced/response-directly/#return-a-response

**Reply by jasonyang101** [reply]
@jgould22 thanks for the response. I actually already have done this using the `ORJSONResponse` class. However, I still see a slowdown of ~7 seconds on a 35MB response - specifically between the statement A and statement B in the above prompt.

I'm unclear on why or how this makes sense - I have pretty extensive logging that makes it seem like I'm doing nothing else on the pod at that time, and this is reproducible.

To frame my question another way: with FastAPI and synchronous endpoints, is it reasonable to attribute this 7 second slowdown to the request size? or is it unlikely that it would take this long just to return a response
